test/test_appdata.py

```diff
@@ -15,6 +15,7 @@ def appdatacheck():
     return output, test
 
 
+@pytest.mark.skipif(not HAS_APPSTREAM_GLIB, reason='Optional dep missing')
```

Comment:
Please put detailed reasons for each `skipif` stanza. We need to know _exactly why_ a test is skipped.